src/Compiler/SyntaxTree/SyntaxTreeOps.fs

```diff
@@ -986,7 +986,10 @@ let (|ParsedHashDirectiveArguments|) (input: ParsedHashDirectiveArgument list) =
     List.map
         (function
         | ParsedHashDirectiveArgument.String(s, _, _) -> s
-        | ParsedHashDirectiveArgument.SourceIdentifier(_, v, _) -> v)
+        | ParsedHashDirectiveArgument.Int32(n, _) -> string n
+        | ParsedHashDirectiveArgument.SourceIdentifier(_, v, _) -> v
+        | ParsedHashDirectiveArgument.Ident(ident, _) -> ident.idText
+        | ParsedHashDirectiveArgument.LongIdent(id, _) -> System.String.Join(".", id.LongIdent |> List.map (fun ident -> ident.ToString())))
```

Comment:
I was checking the new tests, but I don't think there is one that covers long idents? iirc, that's `a.b.c`, right?

Reply:
There isn't a test for long ident, because we don't have a feature that relies on it.  Amusingly my main motivator for "wasting" time on this was the feedback to this: https://github.com/dotnet/fsharp/pull/17140

> I think using a hash directive for this makes the most sense but as we use the normal parser for them it also means we have to wrap the expression in parentheses. I hope that's an acceptable tradeoff.

I think: ```` #help String.filter ```` is a lot better than ````fsi.h  String.filter````

Reply:
> I think: #help String.filter is a lot better than fsi.h  String.filter

yep, couldn't agree more!